### ISSUE TITLE
🔗 Linkdump: LLMs

### ISSUE THREAD

**Issue by garritfra** [garritfra/garrit.xyz#518]
The more I'm getting into large language models (LLMs), the more I'm fascinated about what you can do with them. To "digest" my reading list of cool articles and projects regarding LLMs, I assembled the following list. If you're also interested but haven't started your journey into this neverending rabbit hole, these may contain some good pointers:

## Running LLMs

- [Ollama](https://ollama.com/) - Host different LLMs locally (extremely easy!)
- [vLLM](https://docs.vllm.ai) - Like Ollama but more scalable and production-ready
- [llamafile](https://github.com/Mozilla-Ocho/llamafile) - LLMs as a single, portable binary

## Building Chatbots & Applications

- [LlamaIndex](https://www.llamaindex.ai/) - Framework for building LLM applications
- [create-llama](https://github.com/run-llama/create-llama/) - LlamaIndex template generator (perfect for if you don't know how to structure your app)
- [LangChain](https://www.langchain.com/) - Like LlamaIndex with a less clean interface IMO
- [Haystack](https://haystack.deepset.ai/overview/intro) - Another LLM framework. Haven't tested it though
- [Vercel AI SDK](https://sdk.vercel.ai/docs/introduction) - For integrating your LLM app into a web frontend

## Training and finetuning

- [HuggingFace](https://huggingface.co/) - GitHub but for LLMs
- [LoRA for Fine-Tuning LLMs explained with codes and example](https://medium.com/data-science-in-your-pocket/lora-for-fine-tuning-llms-explained-with-codes-and-example-62a7ac5a3578) - Great LoRA Primer
- [AutoTrain](https://huggingface.co/autotrain) - NoCode LLM finetuning
- [llm-toolkit](https://github.com/georgian-io/LLM-Finetuning-Toolkit) - Another finetuning framework
- [MLFlow](https://mlflow.org/) - End to End platform for LLM training

## Miscellaneous projects

- [llama-fs](https://github.com/iyaja/llama-fs) - Self-organinizing filesystem
- [paramount](https://github.com/ask-fini/paramount) - Measure agent accuracy

## Blogs

- [Simon Willison](https://simonwillison.net/tags/llms/) - Awesome posts from a LLM enthusiast
- [Matt Richard](https://matt-rickard.com/tags/ai) - Matt is currently inactive, but his blog is a treasure trove
